hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -712,8 +707,14 @@ public void terminate(String reason, Exception cause, boolean clearMetrics,
           worker.entryReader.interrupt();
         }
       }
+      //If worker is already stopped but there was still entries batched,
+      //we need to clear buffer used for non processed entries
+      worker.clearWALEntryBatch();
     }
 
+    if (this.replicationEndpoint != null) {
+      this.replicationEndpoint.stop();
```

Comment:
If move "stop" here, the HBASE-25117 problem still exist?

Reply:
Nope, you are right. This ended up here while resolving the conflicts. Let me move it prior to the